docs: update readme (#50)
dashevo/mn-bootstrap: README.md

# MN Bootstrap

[![standard-readme compliant](https://img.shields.io/badge/readme%20style-standard-brightgreen.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)

> > Distribution package for Dash Masternode installation

Author: Why are we formatting this line as a blockquote? We aren't quoting a person here, and the [Standard Readme spec states](https://github.com/RichardLitt/standard-readme/blob/master/spec.md#short-description) that the short description must not start with `>`.

> > Distribution package for Dash Masternode installation

Author: Why are we formatting this line as a blockquote? We aren't quoting a person here, and the [Standard Readme spec states](https://github.com/RichardLitt/standard-readme/blob/master/spec.md#short-description) that the short description must not start with `>`.

> > Distribution package for Dash Masternode installation

Author: Why are we formatting this line as a blockquote? We aren't quoting a person here, and the [Standard Readme spec states](https://github.com/RichardLitt/standard-readme/blob/master/spec.md#short-description) that the short description must not start with `>`.

Reviewer: My bad. It was required before https://github.com/dashevo/drive/blob/v0.13-dev/README.md

  -p, --operator-private-key=operator-private-key  operator private key
```

To start a masternode for Evonet:

```bash
$ mn start evonet 1.2.3.4 20001 -p 2058cd87116ee8492ae0db5d4f8050218588701636197cfcd124dcae8986d514
```

To start a full node for Evonet:

```bash
$ mn start evonet 1.2.3.4 19999 -f
```

### Stop node

The `stop` command is used to stop a running node.

```
USAGE
  $ mn stop PRESET
ARGUMENTS
  PRESET  (local|testnet|evonet) preset to use
```

To stop an Evonet node:

```bash
$ mn stop evonet
```

### Register masternode

The `register` command creates a collateral funding transaction and then uses it to register a masternode on the specified network. It does not configure or start a masternode on the host.

#### Funding collateral

Before registering the masternode, you must send more than 1000 Dash to an address on the network you intend to use. 1000 Dash is used for the collateral transaction, and the remainder will be used for transaction fees. Make sure you have access to the private key for this address, since you will need to provide it in the next step. If using Dash Core, you can get the private key for a given address using the following command:

> 

Reviewer: Don't we need to describe `mn wallet generate-to-address` here as an option for the local preset?

Author: Done, please review :)

Reviewer: Looks good! 👍

# MN Bootstrap

[![standard-readme compliant](https://img.shields.io/badge/readme%20style-standard-brightgreen.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)

Distribution package for Dash Masternode installation

## Table of Contents

> - [Pre-requisites](#pre-requisites)

Reviewer: Do we still have this section?

Reviewer 2: No, looks like that section was removed. I deleted it in b2ea9c6.

[![standard-readme compliant](https://img.shields.io/badge/readme%20style-standard-brightgreen.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)

> Distribution package for Dash Masternode installation

## Table of Contents

- [Install](#install)
- [Usage](#usage)

>   - [Start nodee](#start-node)

Reviewer 2: ```suggestion
  - [Start node](#start-node)
```

  - [Stop node](#stop-node)

>   - [Register masternode](#register-masternonde)

Reviewer 2: ```suggestion
  - [Register masternode](#register-masternode)
```

 - Evonet - masternode with Evonet configuration
 - Testnet - masternode with testnet configuration

There are two ways to apply a preset:

>  1. Rename corresponding dotenv file (i.e. `.env.local`) to `.env`

Reviewer: I would move to a separate section like how to use `docker-compose`.

Author: Actually should we be using `docker-compose` at all anymore? If you start a masternode with `mn start [args]` it will ignore any settings (like the BLS privkey) made in the `.env` file, so should we remove documentation on starting with `docker-compose` completely? Or should I document in separate sections how to do configuration and startup with both the CLI and docker-compose + .env files?

Reviewer: Yeah, I think we need to add a separate section at the bottom how to use docker-compose directly. It could be useful if you need to check logs and so on.

Reviewer: Talking about sections:

We follow usally https://github.com/RichardLitt/standard-readme

Docker, downloading repo and npm install should go to Install section.

Presets explanation, start, stop, reset, masternode registration, using via docker-compose should go to Usage

Author: Cool, I had no idea about standard readme until now. I have applied the specification, and documented the CLI and Docker Compose steps separately, let me know what you think.

 2. Add `--env-file` option to `docker-compose` command

Edit your chosen preset file to specify configuration variables before starting the masternode.

### Start

> In order to run a masternode use Docker Compose:

Reviewer: Should we describe here how to start a full node or masternode for specific preset?

Also, I would describe how to register masternode.

Author: Added syntax and examples for individual commands, is this what you had in mind?

Reviewer: Not really, please check the comments above.

$ docker-compose up
```

### CLI

A CLI is available to perform routine tasks. It can be invoked in two ways:
 1. From the repository folder with `node bin/mn`
 2. By creating a global link with `sudo npm link`, then invoking with `mn`

To list available commands, either run `mn` with no parameters or execute `mn help`. To list the help on any command just execute the command, followed by the `--help` option.

```
USAGE
  $ mn [COMMAND] [ARGS]

COMMANDS
  help      display help for mn
  register  Register masternode
  reset     Reset masternode data
  start     Start masternode

>   status    Show masternode status

Reviewer 2: Should not merge before #49 as this command does not exist yet.

  PRESET  (local|testnet|evonet) preset to use
```

> To stop a an Evonet node:

Reviewer 2: ```suggestion
To stop an Evonet node:
```

$ mn register testnet cVdEfkXLHqftgXzRYZW4EdwtcnJ8Mktw9L4vcEcqbVDs3e2qdzCf 1.2.3.4 19999
```

#### Reset data

> The `reset` command removes all data corresponding to specified preset and allow you to start node from the scratch.

Reviewer 2: ```suggestion
The `reset` command removes all data corresponding to the specified preset and allows you to start a node from scratch.
```
